modules/programs/boxxy.nix

```diff
@@ -0,0 +1,126 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.boxxy;
+
+  boxxyRulesOpts = types.submodule {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          Unique identifier of the boxxy rule. This can be any single-line string.
+        '';
+      };
+
+      target = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "~/.ssh";
+        description = ''
+          What directory/file to redirect.
+        '';
+      };
+
+      rewrite = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "$XDG_CONFIG_HOME/ssh";
+        description = ''
+          Where that file/directory should be rewritten to.
+        '';
+      };
+
+      mode = mkOption {
+        type = types.enum [
+          "file"
+          "directory"
+        ];
+        default = "directory";
+        description = ''
+          Does the current path redirect a file or a directory?
+        '';
+      };
+
+      only = mkOption {
+        type = types.listOf (types.str);
+        default = [];
+        example = literalExpression ''
+          [
+            "bash"
+            "/usr/bin/sh"
+          ]
+        '';
+        description = ''
+          Apply redirection ONLY to specified executable name.
+        '';
+      };
+
+      context = mkOption {
+        type = types.listOf (types.str);
+        default = [];
+        example = [
+          "/home/example/Projects/my-project"
+        ];
+        description = ''
+          Apply redirection ONLY when in a certain directory.
+        '';
+      };
+
+      env = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExpression ''
+          {
+            MY_ENV_VAR = "my_env_var_value";
+          }
+        '';
+        description = ''
+          Give certain environment variables for said match.
+        '';
+      };
+    };
+  };
+in {
+  options.programs.boxxy = {
+    configPath = mkOption {
+      default = "${config.xdg.configHome}/boxxy/boxxy.yaml";
+      defaultText = "$XDG_CONFIG_HOME/boxxy/boxxy.yaml";
+      type = types.str;
+      description = "Path where to put the generated configuration file";
+    };
```

Comment:
Do we need this option? cf https://github.com/nix-community/home-manager/pull/4017#discussion_r1215390589

Reply:
I think it would be better to remove it then